### ISSUE TITLE
[T2]: `test_posttest.py::test_collect_techsupport` causing OOM Kernel Panics.

### ISSUE THREAD

**Issue by arista-nwolfe** [sonic-net/sonic-mgmt#15152]
### Issue Description

When running `test_posttest.py::test_collect_techsupport` we can see kernel panics due to the system running out of memory.
```
root@cmp206-5:~# grep -i "kernel panic" /var/crash/202410172154/dmesg.202410172154
[12183.421193] Kernel panic - not syncing: Out of memory: compulsory panic_on_oom is enabled
```

Throughout the whole test the free memory was very low:
```
Every 2.0s: free -m

               total        used        free      shared  buff/cache   available
Mem:           15838       13390         243         584        3130        2448
Swap:              0           0           0
```

I monitored `top` while the test was running and I saw a few processes consume a bit of memory throughout the test:
```
1223288 root      25   5   73368  51968  18668 R  11.6   0.3   0:00.35 sonic-db-dump
1204423 root      20   0  103208  75068  23756 S   0.0   0.5   0:00.54 show
```

But then near the end of the test a `python` process was spawned which consumed a very large amount of memory ~2gb which pushed us over the limit:
```
1237472 root      20   0 1916700   1.8g  11712 S   0.3  11.7   0:04.13 python
```

### Results you see

Kernel panic due to OOM

### Results you expected to see

No kernel panics

### Is it platform specific

generic

**Comment by arista-nwolfe** [comment]
Note that I don't see OOM kernel panics when manually running `show techsupport --since yesterday` on the device.
Lots of memory still gets consumed but no crash, and I didn't see the `python` process spawned at the end that consumed 2GB so possibly this is related to the `fetch` vs `fetch_no_slurp` linked in this issue.

**Comment by auspham** [comment]
Hi @arista-nwolfe, Does this problem still occurs after the fix in #15145?

**Comment by arista-nwolfe** [comment]
> Hi @arista-nwolfe, Does this problem still occurs after the fix in #15145?

I wasn't able to reproduce the kernel panic after this fix merged, closing this bug.
